src/unix/async.c

```diff
@@ -179,6 +183,15 @@ static void uv__async_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
 
 
 static void uv__async_send(uv_loop_t* loop) {
+#if UV__KQUEUE_EVFILT_USER
+  int err;
+  struct kevent ev;
+
+  EV_SET(&ev, loop->async_io_watcher.fd, EVFILT_USER, 0, NOTE_TRIGGER, 0, 0);
+  err = kevent(loop->backend_fd, &ev, 1, NULL, 0, NULL);
```

Comment:
Out of curiosity, would it be any simpler / faster to register it as an EVFILT_TIMER that is available everywhere, like so? (with corresponding adjustment to `ev->filter` cases to handle it)

```suggestion
  EV_SET(&ev, loop->async_io_watcher.fd, EV_ADD | EVFILT_TIMER | EV_ONESHOT, 0, 0, 0, 0);
  err = kevent(loop->backend_fd, &ev, 1, NULL, 0, NULL);
```

Reply:
I have no idea. AFAIK those renowned cross-platform network frameworks rarely use `EVFILT_TIMER` as the notification mechanism for `kqueue`, they usually use `EVFILT_USER` which is also recommended to be used as a wakeup event. Besides, if `EV_ONESHOT` is set, we have to rearm that event after we consume it, it doesn't look like a pragmatic event for the notification mechanism for `kqueue`.